snowshu/adapters/source_adapters/snowflake_adapter.py

```diff
@@ -252,7 +262,11 @@ def _build_conn_string(self, overrides: Optional[dict] = {}) -> str:
         get_string = "?" + "&".join([arg for arg in get_args])
         return (''.join(conn_parts)) + get_string
 
-    def get_relations_from_database(self, database: str) -> List[Relation]:
+    @overrides
+    def _get_relations_from_database(self, schema_obj: BaseSourceAdapter._DatabaseObject) -> List[Relation]:
+        quoted_database = schema_obj.full_relation.quoted(schema_obj.full_relation.database)  # quoted db name
```

Comment:
this is much much cleaner at the adapter level. We may eventually want to roll this into the base adapter and param the quoting rules once we have several adapters? but this is much more scalable

Reply:
Yep, this is one of the pieces that should be moved out of the Relation object and into the adapters.

Should I add issues for those so we have record of them (this and the decoupling of pattern matching)?